typed-racket-test/unit-tests/static-contract-optimizer-tests.rkt

```diff
@@ -352,28 +355,28 @@
 
     ;; more Or case
     (check-optimize
-      ;; (or (or ....)), both "or"s contain non-flat contracts --- don't optimize
+      ;; flatten & de-duplicate nested or/sc
       (or/sc cons?/sc (or/sc cons?/sc (box/sc cons?/sc)) (box/sc cons?/sc))
-      #:pos (or/sc cons?/sc (or/sc cons?/sc (box/sc cons?/sc)) (box/sc cons?/sc))
-      #:neg (or/sc cons?/sc (or/sc cons?/sc (box/sc cons?/sc)) (box/sc cons?/sc)))
-    (check-optimize
-      ;; (or (or ...)), only the inner "or" contains a non-flat contract --- don't optimize
-      (or/sc cons?/sc (or/sc cons?/sc (box/sc cons?/sc)))
-      #:pos (or/sc cons?/sc (or/sc cons?/sc (box/sc cons?/sc)))
-      #:neg (or/sc cons?/sc (or/sc cons?/sc (box/sc cons?/sc))))
-    (check-optimize
-      ;; (or (or ...)), only the outer "or" contains a non-flat contract --- still don't optimize
-      (or/sc (box/sc cons?/sc) (or/sc cons?/sc set?/sc))
-      #:pos (or/sc (box/sc cons?/sc) (or/sc cons?/sc set?/sc))
-      #:neg (or/sc (box/sc cons?/sc) (or/sc cons?/sc set?/sc)))
+      #:pos (or/sc (box/sc cons?/sc) cons?/sc)
+      #:neg (or/sc (box/sc cons?/sc) cons?/sc))
+    (let ([rec-chaperone (recursive-sc (list #'x) (list cons?/sc) (box/sc (recursive-sc-use #'x)))])
```

Comment:
Now using `recursive-sc` instead of `or/sc` because a recursive contract doesn't correspond to a value constructor (like `or/sc`) and doesn't get collapsed by other optimizations (unlike `or/sc`)